committeeoversightapp/management/commands/import_data.py

```diff
@@ -115,98 +110,165 @@ def add_house_hearings(self):
 
-                    row_count += 1
+                if (committee1 and not subcommittee1) or (subcommittee1 == (committee1 + str(0))):
+                    committees_filtered.append(committee1)
+                elif committee1:
+                    committees_filtered.append(subcommittee1)
+
+                if (committee2 and not subcommittee2) or (subcommittee2 == (committee2 + str(0))):
+                    committees_filtered.append(committee2)
+                elif committee2:
+                    committees_filtered.append(subcommittee2)
+
+                try:
+                    event = Event.objects.get(
+                                            name=name,
+                                            start_date=start_date,
+                                            jurisdiction_id=self.jurisdiction_id,
+                                            classification=classification
+                                            )
+                except ObjectDoesNotExist:
+                    if name:
+                        self.smart_match(matched, name, committees_filtered, start_date, category, source, classification)
 
     def add_senate_hearings(self):
         with open('data/final/senate.csv', 'r') as csvfile:
             reader = csv.DictReader(csvfile)
-            row_count = 2
+            self.row_count = 2
 
             for row in reader:
 
-                source = row['source']
-                start_date = row['Date'].split('T', 1)[0]
+                matched = False
+
                 name = row['Hearing/Report']
+                start_date = row['Date'].split('T', 1)[0]
+                source = row['source']
                 classification = row['Type']
-                committees = [row['Committee1'], row['Committee2']]
                 category = row['Category1']
+                hearing_number_raw = row['Hearing #']
 
-                if name:
-                    # get or create hearing
-                    event, created = Event.objects.get_or_create(jurisdiction_id = self.jurisdiction_id,
-                                    name = name,
-                                    start_date = start_date,
-                                    classification = classification)
+                committees = [row['Committee1'], row['Committee2']]
+                committees_filtered = [committee for committee in committees if committee]
 
+                try:
+                    event = Event.objects.get(
+                                            name=name,
+                                            start_date=start_date,
+                                            jurisdiction_id=self.jurisdiction_id,
+                                            classification=classification
+                                            )
+                except ObjectDoesNotExist:
+                    # try to match by serial number
+                    if name and hearing_number_raw:
```

Comment:
I think we want to try to match by hearing_number first and then fall back to name and date.

Reply:
This is checking that the hearing wasn't already imported from the spreadsheet, so that this script can be rerun. Do you think hearing_number should still be checked first?

Reply:
if you want to check to see if an item has already been linked, then maybe also check for source?

Reply:
Good call